ros_cross_compile/data_collector.py

```diff
@@ -40,15 +44,20 @@ class Units(Enum):
 class DataCollector:
     """Provides an interface to collect time series data."""
 
-    def __init__(self):
+    def __init__(self, print_metrics):
         self._data = []
+        self._print_metrics = print_metrics
 
     def add_datum(self, new_datum: Datum):
         self._data.append(new_datum)
 
     def serialize_data(self) -> List[Dict]:
         return list(map(lambda d: d._asdict(), self._data))
 
+    def print_data(self, units: str, metric: Datum):
```

Comment:
Let's do some form of pretty-printing to make sure the data is readable. I am basing my understanding of how this feature should work based on Ansible's `profile_tasks`. Here's a sample output from the end of such a run. The formatting doesn't need to be exactly this, but something along these lines

```
Wednesday 10 January 2018  11:29:17 +0530 (0:00:00.765)       0:00:07.370 ***** 
=============================================================================== 
TASK: setup ------------------------------------------------------------- 6.00s
TASK: Test the user access ---------------------------------------------- 0.77s
TASK: copy ssh public key to remote host -------------------------------- 0.29s
TASK: create user on remote machine ------------------------------------- 0.22s
```